### PR TITLE
Configure orval to add network parameter for controlling baseURL

### PR DESCRIPTION
**Pull request by lukaw3d**
~Includes changes from https://github.com/oasisprotocol/explorer/pull/472~
Supersedes https://github.com/oasisprotocol/explorer/pull/441 

Related to https://github.com/oasisprotocol/explorer/pull/404#discussion_r1200771339

### CODE REVIEW

File: src/oasis-indexer/api.ts [hunk 1]
```diff
@@ -122,19 +92,20 @@ function arrayify<T>(arrayOrItem: null | undefined | T | T[]): T[] {
...
+  params?,
+  options?,
+) => {
   const ticker = getTickerForNetwork(network)
-  return generated.useGetConsensusTransactions(params, {
+  return generated.useGetConsensusTransactions(network, params, {
     ...options,
     query: {
       ...options?.query,
       queryKey: [`/${network}/consensus/transactions`, ...(params ? [params] : [])],
```
Comment (csillag) [inline src/oasis-indexer/api.ts:105]: Fixing the `queryKey` should be automatic now, right? (Here and a bunch of other places, too)

File: src/oasis-indexer/api.ts [hunk 2]
```diff
@@ -151,25 +122,27 @@ export const useGetConsensusTransactions: typeof _f1 = (network, params?, option
...
+  params?,
+  options?,
+) => {
   const ticker = getTickerForNetwork(network)
-  return generated.useGetRuntimeTransactions(runtime, params, {
+  return generated.useGetRuntimeTransactions(network, runtime, params, {
     ...options,
     query: {
       ...options?.query,
       queryKey: [`/${network}/${runtime}/transactions`, ...(params ? [params] : [])],
```
Comment (csillag) [inline src/oasis-indexer/api.ts:142]: Again, useless `queryKey` override

File: src/oasis-indexer/api.ts [hunk 3]
```diff
@@ -189,25 +162,26 @@ export const useGetRuntimeTransactions: typeof _f2 = (network, runtime, params?,
...
+  txHash,
+  options?,
+) => {
   const ticker = getTickerForNetwork(network)
-  return generated.useGetConsensusTransactionsTxHash(txHash, {
+  return generated.useGetConsensusTransactionsTxHash(network, txHash, {
     ...options,
     query: {
       ...options?.query,
       queryKey: [`/${network}/consensus/transactions/${txHash}`],
```
Comment (csillag) [inline src/oasis-indexer/api.ts:181]: queryKey... etc

File: src/oasis-indexer/api.ts [hunk 4]
```diff
@@ -122,19 +92,20 @@ function arrayify<T>(arrayOrItem: null | undefined | T | T[]): T[] {
...
     query: {
       ...options?.query,
       queryKey: [`/${network}/consensus/transactions`, ...(params ? [params] : [])],
     },
-    axios: {
-      ...options?.axios,
-      ...getBaseUrlParamFor(network),
+    request: {
+      ...options?.request,
       transformResponse: [
```
Comment (csillag) [inline src/oasis-indexer/api.ts:109]: Interesting, so is the `transformResponse` accepted both as an option inside `axios` and in `request` ?

**Comment by csillag** [comment]
> Supersedes and closes https://github.com/oasisprotocol/explorer/pull/441

Not completely, because #441 has some more goodies that can be added on top of this, just not everything has been pushed yet.

**Comment by csillag** [comment]
> Supersedes and closes https://github.com/oasisprotocol/explorer/pull/441

Not completely, because #441 has some more goodies that can be added on top of this, just not everything has been pushed yet.

**Review by csillag — APPROVED** [review]
Reviewed. LGTM, except a few small questions.

File: src/oasis-indexer/api.ts (hunk 1 above)
Comment (lukaw3d) [reply src/oasis-indexer/api.ts:105]: yup! somehow missed them all

File: src/oasis-indexer/api.ts (hunk 4 above)
Comment (lukaw3d) [reply src/oasis-indexer/api.ts:109]: When customizing orval it changed the field name from `axios` to `request` (`query` field still doesn't accept it)

**Pull request merged**